force-app/tdtm/classes/TDTM_TriggerHandler.cls

```diff
@@ -40,6 +40,8 @@ public class TDTM_TriggerHandler {
     
     private static Boolean defaultRecordsInserted = false;
     
+    private static Boolean preventCachingFalseTransaction = false;
```

Comment:
End up having to do a static variable to preserve the flag from Before to After Context. The `addError` is called in Before Context. However, the caching is happening in the after context.

Reply:
I think we can improve the name for this variable a bit, couple of ideas:

* skipCachingNewList
* isTransactionRollingBack